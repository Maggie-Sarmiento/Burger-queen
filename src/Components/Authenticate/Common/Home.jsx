/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { Outlet, useLocation } from 'react-router-dom';
import Sidebar from './Sidebar';
import Header from './Header';
import styles from './Home.module.css';
import Welcome from './Welcome';

const Home = ({ role, authenticate }) => {
  const [count, setCount] = useState(0);
  const [dataProduct, setDataProduct] = useState([]);
  const location = useLocation();
  console.log(location);

  if (role === 'Mesero') {
    return (
      <section className={styles.waiter}>
        <section className={styles.sidebar}>
          <Sidebar role={role} />
        </section>
        <main className={styles.main}>
          <Header
            role={role}
            authenticate={authenticate}
            count={count}
            dataProduct={dataProduct}
            setDataProduct={setDataProduct}
            setCount={setCount}
            className={styles.profile}
          />
          <Outlet context={[count, setCount, dataProduct, setDataProduct]} />
        </main>
      </section>
    );
  }

  return (
    <section className={styles.admin}>
      <section className={styles.sidebar}>
        <Sidebar role={role} />
      </section>
      <main className={styles.main}>
        <Header authenticate={authenticate} className={styles.profile} />
        <div className={styles.adminOutlet}>
          {location.pathname === '/home' ? <Welcome /> : <Outlet />}
        </div>
      </main>
    </section>
  );
};

export default Home;
