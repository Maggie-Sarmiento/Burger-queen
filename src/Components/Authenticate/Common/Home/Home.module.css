.admin {
  display: flex;
}

.sidebar {
  width: 10%;
  position: fixed;
  top: 0;
  bottom: 0;
}

.main {
  display: flex;
  flex-direction: column;
  width: 75%;
  margin-left: 15%;
  overflow-x: hidden;
}

/* .adminOutlet {
  height: 80vh;
} */

.welcome {
  margin-top: 8em;
}

@media (min-width: 540px) {
  .sidebar{
    width: 10%;
  }
}

@media (min-width: 1280px) {
}

@media (min-width: 1900px) {
  .main {
    width: 86%;
    margin-left: 10%;
  } 
}
