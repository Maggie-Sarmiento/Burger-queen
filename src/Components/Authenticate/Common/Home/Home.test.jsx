/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import Home from './Home';

describe('test Home', () => {
  const mockAuthenticate = jest.fn();
  test('render component Home', async () => {
    let root;
    act(() => {
      root = create(
        <Router>
          <Home authenticate={mockAuthenticate} />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
  test('render component header', async () => {
    const value = 'Mesero';
    render(
      <Router>
        <Home role={value} authenticate={mockAuthenticate} />
      </Router>,
    );
    const button = screen.getByText(/Menu Desayuno/i);
    expect(button).toBeInTheDocument();
  });
});
