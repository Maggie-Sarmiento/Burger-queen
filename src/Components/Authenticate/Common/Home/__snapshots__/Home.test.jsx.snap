// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home render component Home 1`] = `
<section
  className="admin"
>
  <section
    className="sidebar"
  />
  <main
    className="main"
  >
    <header
      className="header"
    >
      <div
        className="Profile"
      >
        <div>
          <button
            aria-haspopup="true"
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root btnProfile css-xa6ayv-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            id="basic-button"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            undefinedundefined
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      className="adminOutlet"
    />
  </main>
</section>
`;
